### ISSUE TITLE
[a11y] Onboarding's Pick a toolbar placement radio buttons are missing programmatic selected state

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#27735]
## Steps to reproduce
1. Have VoiceOver running
1. Open Firefox for the first time
2. On the onboarding screens, locate a "Pick a toolbar placement" screen (5 of 5 in the onboarding flow)
3. Locate "Top" and "Bottom" radio buttons and observe the VO announcement, esp. for the visually selected one

### Expected behavior
The visually selected theme would be announced as such by VO, i.e. "Bottom, button, selected"

### Actual behavior
The toolbar placement picker radio buttons are not announced as "selected" even when they visually are. Users of VoiceOver would have no way to learn which one is, in fact, selected, if any..

### User feedback:
> When I select a button or link, nothing happens
> On the toolbar placement screen, voiceover gives me no indication whether the toolbar option I clicked on was selected. This makes it difficult for me to know whether my preferred toolbar option was chosen. Voiceover should give me verbal indication that the option I clicked on is now selected so I know that I have chosen the correct option.

### Device & build information
Screenshot captured from a user feedback received on July 2, 2025, thus details on the build and iOS version are not known
* Device: ? iPhone
* Build version: (current release)
* First seen version: ?
### Notes
Attachments: 

<img width="474" alt="Bottom toolbar placement is visually selected but not announced as such with VO" src="https://github.com/user-attachments/assets/685c640a-aef8-43e0-b093-99c88f8c4ebe" />

Follow [the Magenta11y Native App > Radio Buttons guide](https://www.magentaa11y.com/#/native-criteria/controls/radio-button?tab=3) for developer notes and testing example and checklist for radio buttons accessibility

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-12738)

**Comment by FilippoZazzeroni** [comment]
cc @thatswinnie @razvanlitianu should we port this or already have on new onboarding ?

**Comment by FilippoZazzeroni** [comment]
those are the other tickets linked to this:
- https://github.com/mozilla-mobile/firefox-ios/issues/27734
- https://github.com/mozilla-mobile/firefox-ios/issues/27731

**Comment by arq0017** [comment]
is it possible to work on this issue ?

**Comment by FilippoZazzeroni** [comment]
Hi @arq0017 sure i'm assigning you. FYI this is part of the Legacy Onboarding if you want more updated tickets please check out [here](https://github.com/mozilla-mobile/firefox-ios/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22Contributor%20OK%22)

**Comment by arq0017** [comment]
I’ve been looking into this and noticed the screen has been updated. Since the issue is no longer reproducible in the latest version, could you confirm if the legacy code has been officially deprecated or removed?

**Comment by FilippoZazzeroni** [comment]
Hi @arq0017 yes i confirm the legacy code is officially deprecated still not removed though. I suggest looking to more recent [tickets](https://github.com/mozilla-mobile/firefox-ios/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22Contributor%20OK%22) thanks very much for looking into it.
